test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java

```diff
@@ -0,0 +1,180 @@
+        }
+    }
+}
+
+public class VThreadEventTest {
+    static final int TCNT1 = 10;
+    static final int TCNT2 = 4;
+    static final int TCNT3 = 4;
+    static final int THREAD_CNT = TCNT1 + TCNT2 + TCNT3;
+    static final long TIMEOUT_BASE = 1_000_000L;
+
+    private static void log(String msg) { System.out.println(msg); }
+
+    private static native int threadEndCount();
+    private static native int threadMountCount();
+    private static native int threadUnmountCount();
+
+    private static volatile int completedNo;
+    private static volatile boolean attached;
+    private static boolean failed;
+    private static List<Thread> threads = new ArrayList(TCNT1);
+
+    public static void main(String[] args) throws Exception {
+        if (Runtime.getRuntime().availableProcessors() < 8) {
+            log("WARNING: test expects at least 8 processors.");
+        }
+        Counter ready1 = new Counter(THREAD_CNT);
+        Counter ready2 = new Counter(THREAD_CNT);
+        Counter mready = new Counter(1);
+
+        try (ExecutorService executorService = Executors.newVirtualThreadPerTaskExecutor()) {
+            for (int tCnt = 0; tCnt < TCNT1; tCnt++) {
+                executorService.execute(() -> {
+                    synchronized (threads) {
+                        threads.add(Thread.currentThread());
+                    }
+                    log("test1 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    try {
+                        // timeout is big enough to keep mounted untill interrupted
+                        Thread.sleep(20000);
+                    } catch (InterruptedException ex) {
+                        // it is expected, ignore
+                    }
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            for (int tCnt = 0; tCnt < TCNT2; tCnt++) {
+                executorService.execute(() -> {
+                    log("test2 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    while (!attached) {
+                        // keep mounted
+                    }
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            for (int tCnt = 0; tCnt < TCNT3; tCnt++) {
+                executorService.execute(() -> {
+                    log("test3 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    while (!attached) {
+                        // keep mounted
+                    }
+                    LockSupport.parkNanos(10 * TIMEOUT_BASE);
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            ready1.await();
+            mready.decr();
+            VirtualMachine vm = VirtualMachine.attach(String.valueOf(ProcessHandle.current().pid()));
```

Comment:
I think sleep is needed here so threads which should be unmounted have time to unmount before attach.

Reply:
Would it makes sense also to check that thread state is TIMED_WAITING. It should be set  TIMED_WAITING unmounted threads https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Thread.State.html#TIMED_WAITING

Reply:
It is strange that the the tested vthreads in sleep(timeout) have sate WAITING, not TIMED_WAITING.
It can be a bug in the implementation.
I've decided to add a short sleep. Checking states looks a little bit over complicated.

Reply:
Could you please also add comment with sleep which describe why is it needed and mention that the insufficient sleep time couldn't cause test failures. Test should pass anyway just don't test expected state.

Reply:
Done.